### PR TITLE
fix(css_formatter): keep @charset double quote in single quote config

### PR DESCRIPTION
**Pull request by fireairforce**
## Summary

closes: #4384 

`@charset` must always have double quotes: https://www.w3.org/TR/css-syntax-3/#determine-the-fallback-encoding

I add a param `StringLiteralParentKind` for `impl FormatLiteralStringToken` to judge if the `CSS String`'s parent is a `CSS_CHARSET_AT_RULE` which represents the syntax like:`@charset`, i refer the same util at `biome_js_formatter`.

if the string is `StringLiteralParentKind::CharsetAtRule`, i will use the double quote for the string in any case, or we will execute the normal logic of `biome_css_formatter`.

## Test Plan

Test case as follows: 

```css
@charset·"UTF-8";
@charset "iso-8859-15";
@charset "any-string-is-okay";
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
 #### Bug fixes
 
 - Fix [#4121](https://github.com/biomejs/biome/issues/4121). Respect line width when printing multiline strings. Contributed by @ah-yu
```
Comment (ematipico) [inline CHANGELOG.md:160]: In this line, it's not clear that the fix is for the CSS files

**Review by ematipico — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (fireairforce) [reply CHANGELOG.md:160]: should i update this here？ 😂，it seems not my PR CHANGELOG，Then I changed it together by the way?

File: CHANGELOG.md [hunk 2]
```diff
@@ -158,6 +158,7 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
...
+- Fix [#4384](https://github.com/biomejs/biome/issues/4384). Keep `@charset` dobule quote under any situation for css syntax rule. Contributed by @fireairforce
```
Comment (fireairforce) [inline CHANGELOG.md:161]: @ematipico here is my changelog, lol

File: CHANGELOG.md (hunk 1 above)
Comment (ematipico) [reply CHANGELOG.md:160]: Oh, my bad! I didn't notice it wasn't you. It's fine then :)

**Pull request merged**